functions/Find-DbaDisabledIndex.ps1

```diff
@@ -0,0 +1,180 @@
+
+        .EXAMPLE
+            Find-DbadisabledIndex -SqlInstance sqlserver2016 -Database db1, db2
+
+            Generates the SQL Statement to to drop selected indexes in databases db1 & db2 on server "sqlserver2016".
+
+        .EXAMPLE
+            Find-DbadisabledIndex -SqlInstance sqlserver2016
+
+            Generates the SQL statements to drop selected indexes on all user databases.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    Param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]
+        $SqlCredential,
+        [Alias("Databases")]
+        [object[]]$Database,
+        [object[]]$ExcludeDatabase,
+        [switch]$NoClobber,
+        [switch]$Append,
+        [switch][Alias('Silent')]$EnableException
+    )
+
+    begin {
+        # 
+        $disabledQuery = "
+        SELECT DB_NAME() AS 'DatabaseName'
+        ,s.name AS 'SchemaName'
+        ,t.name AS 'TableName'
+        ,i.object_id AS ObjectId
+        ,i.name AS 'IndexName'
+        ,i.index_id as 'IndexId'
+        ,i.type_desc as 'TypeDesc'
+        FROM sys.tables t
+        JOIN sys.schemas s
+            ON t.schema_id = s.schema_id
+        JOIN sys.indexes i
+            ON i.object_id = t.object_id
+        WHERE i.is_disabled = 1"
+
+        Write-Message -Level Output -Message "Attempting to connect to Sql Server."
+        $server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
+    }
+    process {
+        if (Test-FunctionInterrupt) { return }
+
+        if ($server.VersionMajor -lt 9) {
+            Stop-Function -Message "This function does not support versions lower than SQL Server 2005 (v9)."
+            return
+        }
+
+        $lastRestart = $server.Databases['tempdb'].CreateDate
+        $endDate = Get-Date -Date $lastRestart
+        $diffDays = (New-TimeSpan -Start $endDate -End (Get-Date)).Days
+
+        <#
+            Validate if server version is:
+                - sql 2012 and if have SP3 CU3 (Build 6537) or higher
+                - sql 2014 and if have SP2 (Build 5000) or higher
+            If the major version is the same but the build is lower, throws the message
+        #>
+        if ($pipedatabase.Length -gt 0) {
+            $database = $pipedatabase.name
+        }
+
+        if ($database.Count -eq 0) {
+            $database = ($server.Databases | Where-Object { $_.IsSystemObject -eq 0 -and $_.IsAccessible}).Name
+        }
+
+        if ($database.Count -gt 0) {
+            foreach ($db in $database) {
+                
+                if ($ExcludeDatabase -contains $db -or $null -eq $server.Databases[$db]) {
+                    continue
+                }
+                if ($server.Databases[$db].IsAccessible -eq $false) {
+                    Write-Message -Level Warning -Message "Database [$db] is not accessible."
```

Comment:
This would seem redundant since we already removed databases that were not accessible. We are not refreshing the server object so even if a database was taken offline before it hits this line we would not know about it.